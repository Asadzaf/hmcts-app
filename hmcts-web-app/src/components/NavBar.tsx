import { NavLink, useNavigate } from "react-router-dom";
import '../assets/govuk-frontend/dist/govuk/govuk-frontend.min.css'


const NavBar = ({
  currentPage = ''
}) => {
  const navigate = useNavigate();

  switch (currentPage) {
    case 'allTasks':
      return allTasksNavbar();
    case 'viewTask':
      return viewTaskNavbar();
    case 'createTask':
      return createTaskNavbar();
    default:
      return null;
  }

  function createTaskNavbar() {
    return (
      <nav className="govuk-breadcrumbs" aria-label="Breadcrumb">
        <ol className="govuk-breadcrumbs__list">
          <li className="govuk-breadcrumbs__list-item">
            <a className="govuk-breadcrumbs__link" onClick={() => navigate('/')}>Home</a>
          </li>
          <li className="govuk-breadcrumbs__list-item">
            <a className="govuk-breadcrumbs__link" onClick={() => navigate('/createTask')}>Create Task</a>
          </li>
        </ol>
      </nav>
    );
  }

  function allTasksNavbar() {
    return (
      <nav className="govuk-breadcrumbs" aria-label="Breadcrumb">
        <ol className="govuk-breadcrumbs__list">
          <li className="govuk-breadcrumbs__list-item">
            <a className="govuk-breadcrumbs__link" onClick={() => navigate("/")}>Home</a>
          </li>
          <li className="govuk-breadcrumbs__list-item">
            <a className="govuk-breadcrumbs__link" onClick={() => navigate("allTasks")}>All Tasks</a>
          </li>
        </ol>
      </nav>
    );
  };

  function viewTaskNavbar() {
    return (
      <nav className="govuk-breadcrumbs" aria-label="Breadcrumb">
        <ol className="govuk-breadcrumbs__list">
          <li className="govuk-breadcrumbs__list-item">
            <a className="govuk-breadcrumbs__link" onClick={() => navigate('/')}>Home</a>
          </li>
          <li className="govuk-breadcrumbs__list-item">
            <a className="govuk-breadcrumbs__link" onClick={() => navigate('/allTasks')}>All Tasks</a>
          </li>
          <li className="govuk-breadcrumbs__list-item">
            <NavLink className="govuk-breadcrumbs__link" to="/viewTask">View or Update Task</NavLink>
          </li>
        </ol>
      </nav>
    );
  };

}

export default NavBar;