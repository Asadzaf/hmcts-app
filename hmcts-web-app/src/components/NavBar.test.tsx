import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import NavBar from './NavBar';
import { MemoryRouter } from 'react-router-dom';


test('renders Header correctly', () => {
  const { asFragment } = render(
    <MemoryRouter><NavBar currentPage='allTasks' /></MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
});

test('renders Header correctly', () => {
  const { asFragment } = render(<MemoryRouter><NavBar currentPage='createTask' /></MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
});
