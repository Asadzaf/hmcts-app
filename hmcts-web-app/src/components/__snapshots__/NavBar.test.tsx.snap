// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header correctly 1`] = `
<DocumentFragment>
  <nav
    aria-label="Breadcrumb"
    class="govuk-breadcrumbs"
  >
    <ol
      class="govuk-breadcrumbs__list"
    >
      <li
        class="govuk-breadcrumbs__list-item"
      >
        <a
          class="govuk-breadcrumbs__link"
        >
          Home
        </a>
      </li>
      <li
        class="govuk-breadcrumbs__list-item"
      >
        <a
          class="govuk-breadcrumbs__link"
        >
          All Tasks
        </a>
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;

exports[`renders Header correctly 2`] = `
<DocumentFragment>
  <nav
    aria-label="Breadcrumb"
    class="govuk-breadcrumbs"
  >
    <ol
      class="govuk-breadcrumbs__list"
    >
      <li
        class="govuk-breadcrumbs__list-item"
      >
        <a
          class="govuk-breadcrumbs__link"
        >
          Home
        </a>
      </li>
      <li
        class="govuk-breadcrumbs__list-item"
      >
        <a
          class="govuk-breadcrumbs__link"
        >
          Create Task
        </a>
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;
